core/thread/credentials.go

```diff
@@ -0,0 +1,98 @@
+package thread
+
+import (
+	"github.com/libp2p/go-libp2p-core/crypto"
+)
+
+// Signature of an ID
+type Signature []byte
+
+// Credentials are used to prove a caller identity, which is needed
+// for thread ACL checks.
+type Credentials interface {
+	// ThreadID must return the signed thread ID.
+	ThreadID() ID
+
+	// Sign must return the public key and signature which can
+	// be used to verify the caller.
+	Sign() (crypto.PubKey, Signature, error)
+}
+
+// DefaultCreds are useful for either threads that don't require
+// credentials or when the private key is available for local signing.
+type DefaultCreds struct {
+	threadID ID
+	privKey  crypto.PrivKey
+}
+
+// NewDefaultCreds returns new default credentials.
+func NewDefaultCreds(id ID, opts ...CredOption) Credentials {
```

Comment:
Important note: Credentials with a signing key / signature are only required for cases where the underlying thread is private, which is (will be) defined in the ACL. If the thread is wide open, anyone can do anything.